### PR TITLE
feat: enable custom wrapper locations

### PR DESCRIPTION
**Pull request by caalador**
Add annotation Scan that can be given
an array of package strings that
will be scanned for ComponentWrappers.

This enabled custom wrappers outside
of the com.vaadin.testbench.component
package.

### CODE REVIEW

File: vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/Scan.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ ElementType.TYPE, ElementType.METHOD })
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Scan {
+
+    String[] value();
```
Comment (mcollovati) [inline vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/Scan.java:21]: Can we add a javadoc to explain that `value` should be an array of package names?

File: vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/Scan.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * For the full License, see <https://vaadin.com/license/cvdl-4.0>.
+ */
+package com.vaadin.testbench.unit;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ ElementType.TYPE, ElementType.METHOD })
```
Comment (mcollovati) [inline vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/Scan.java:17]: Do we support method level annotation?

File: vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/Scan.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Scan {
```
Comment (knoobie) [inline vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/Scan.java:19]: The name could be anything or nothing.. could this be more elaborative like `@TestBenchUIScan` if it meant to be used by developer to specify their custom wrapper packages.

File: vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/Scan.java (hunk 2 above)
Comment (caalador) [reply vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/Scan.java]: No that was a test, but it didn't really work. Removed

File: vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/Scan.java (hunk 1 above)
Comment (caalador) [reply vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/Scan.java]: Added javadocs

File: vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/Scan.java (hunk 3 above)
Comment (caalador) [reply vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/Scan.java]: TestBenchUIScan sounds like it scans for UI implementations.
Renamed to `@ComponentWrapPackages`

File: vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/ComponentWrapPackages.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+package com.vaadin.testbench.unit;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation to use to scan given packages for component wrappers outside the
+ * default "com.vaadin.flow.component`.
```
Comment (mcollovati) [inline vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/ComponentWrapPackages.java:19]: minor: package has double quote as prefix and backtick as suffix. Maybe we can use `@code` or `@literal` to wrap the package name.
```suggestion
 * default {@code com.vaadin.flow.component}.
```

**Review by mcollovati — APPROVED** [review]

**Pull request merged**